packages/vision/src/index.js

```diff
@@ -410,7 +412,9 @@ Vision.prototype.detect = function(images, options, callback) {
 
       var detections = foundImages
         .map(groupDetectionsByImage)
-        .map(removeExtraneousAnnotationObjects)
+        .map(function(image) {
+          return curation.removeExtraneousAnnotations(image, options.types);
```

Comment:
As far as how to make it more obvious, most likely I will just use an intervening variable. Even just doing:

```javascript
var detections = foundImages.map(groupDetectionsByImage);
detections = dections.map(...).map(...) // and so on
```

...would actually help clarity a lot.